addons/interaction/functions/fnc_passMagazine.sqf

```diff
@@ -35,18 +36,11 @@ private _magToPassIndex = 0;
     };
 } foreach _filteredMags;
 
-//remove all magazines and add them again, except the one to be passed
-//needed because of missing commands, see http://feedback.arma3.com/view.php?id=12782
+//remove the magazine from _player and add it to _target
 _magToPass params ["_magToPassClassName", "_magToPassAmmoCount"];
-_player removeMagazines _magToPassClassName;
-{
-    _x params ["_className", "_ammoCount"];
-    if ((_className == _magToPassClassName) && (_forEachIndex != _magToPassIndex)) then {
-        _player addMagazine [_className, _ammoCount];
-    };
-} foreach _filteredMags;
+[_player, _magToPassClassName, _magToPassAmmoCount] call CBA_fnc_removeMagazine;
```

Comment:
ACE function is better than CBA one, as it's lighter and gives a return
```suggestion
// Exit if failed to remove specific magazine
if !([_player, _magToPassClassName, _magToPassAmmoCount] call EFUNC(common,removeSpecificMagazine)) exitWith {};
```

Reply:
Cool, didn't know that existed 👍